src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadMetadata.java

```diff
@@ -0,0 +1,180 @@
+        }
+        meanBasesPerTemplate = input.readInt();
+    }
+
+    private void serialize( final Kryo kryo, final Output output ) {
+        output.writeInt(contigNameToID.size());
+        for ( final Map.Entry<String, Short> entry : contigNameToID.entrySet() ) {
+            kryo.writeObject(output, entry.getKey());
+            output.writeShort(entry.getValue());
+        }
+        output.writeInt(readGroupToFragmentStatistics.size());
+        for ( final Map.Entry<String, ReadGroupFragmentStatistics> entry : readGroupToFragmentStatistics.entrySet() ) {
+            kryo.writeObjectOrNull(output, entry.getKey(), String.class);
+            kryo.writeObject(output, entry.getValue());
+        }
+        output.writeInt(meanBasesPerTemplate);
+    }
+
+    public short getContigID( final String contigName ) {
+        final Short result = contigNameToID.get(contigName);
+        if ( result == null ) throw new GATKException("No such contig name: "+contigName);
+        return result;
+    }
+
+    public ReadGroupFragmentStatistics getStatistics( final String readGroupName ) {
+        final ReadGroupFragmentStatistics stats = readGroupToFragmentStatistics.get(readGroupName);
+        if ( stats == null ) throw new GATKException("No such read group name: "+readGroupName);
+        return stats;
+    }
+
+    public int getMaxMedianFragmentSize() {
+        return readGroupToFragmentStatistics.entrySet().stream()
+                .mapToInt(entry -> Math.round(entry.getValue().getMedianFragmentSize()))
+                .max()
+                .orElse(0);
+    }
+
+    public int getMeanBasesPerTemplate() { return meanBasesPerTemplate; }
+
+    @Override
+    public boolean equals( final Object obj ) {
+        if ( !(obj instanceof ReadMetadata) ) return false;
+        final ReadMetadata that = (ReadMetadata)obj;
+        return this.contigNameToID.equals(that.contigNameToID) &&
+                this.readGroupToFragmentStatistics.equals(that.readGroupToFragmentStatistics) &&
+                this.meanBasesPerTemplate == that.meanBasesPerTemplate;
+    }
+
+    @Override
+    public int hashCode() {
+        return 47*(47*(47*contigNameToID.hashCode() + readGroupToFragmentStatistics.hashCode()) + meanBasesPerTemplate);
+    }
+
+    public static final class Serializer extends com.esotericsoftware.kryo.Serializer<ReadMetadata> {
+        @Override
+        public void write( final Kryo kryo, final Output output, final ReadMetadata readMetadata ) {
+            readMetadata.serialize(kryo, output);
+        }
+
+        @Override
+        public ReadMetadata read( final Kryo kryo, final Input input, final Class<ReadMetadata> klass ) {
+            return new ReadMetadata(kryo, input);
+        }
+    }
+
+    @DefaultSerializer(ReadGroupFragmentStatistics.Serializer.class)
+    public static final class ReadGroupFragmentStatistics {
+        private final float medianFragmentSize;
+        private final float medianFragmentSizeVariance;
```

Comment:
I think we'll want to compute in terms of median absolute deviations (MADs) -- can you rename this to medianAbsoluteDeviationFragmentSize?